connection.js

```diff
@@ -351,7 +351,9 @@ class Connection {
             }
             else {
                 // unrecognized command
-                plugins.run_hooks('unrecognized_command', this, [methodMatch, remainingMatch]);
+                matches.splice(0,1);
+                matches.splice(1,1);
+                plugins.run_hooks('unrecognized_command', this, matches);
```

Comment:
I don't like the splice here - seems potentially fragile. We should just use `[matches[1], matches[2]]`.

Reply:
There's already named variables right above telling what matches[1] and matches[3] are (not matches[2], using splice isn't easy to intuit), so better would be using [method, remaining] as it was before.

Reply:
The variables are different since `method` is actually `cmd_${methodMatch.toLowerCase()}`.